scicamera/camera.py

```diff
@@ -34,6 +34,14 @@
 class CameraManager:
     cameras: Dict[int, Camera]
 
+    _instance: CameraManager = None
+
+    @classmethod
+    def singleton(cls):
+        if cls._instance is None:
+            cls._instance = cls()
+        return cls._instance
```

Comment:
I always have a little trouble reasoning about the singleton/borg pattern in python. I see some threading stuff here, and IIRC there are some demons there. For example, I'm not sure that you'll get a singleton if the camera manager is imported inside of a thread. Heres a test that might elucidate things:

```python
def thread_function():
    from scicamera.camera import CameraManager
    return CameraManager.singleton().id

# Now launch 100 threads and see if they all have the same `id`

executor = ThreadPoolExecutor()
for singleton_id in executor.map([thread_function] * 100):
    print(f"Singleton id is {singleton_id}")

```